File: unity/Assets/AvalonShell/AvalonLauncherActivity.cs
// AVALON launcher — Unity 6 post-processor forces the stock UnityPlayerActivity to
// android:enabled="false"; the sanctioned entry-point pattern is a custom subclass
// declared as the MAIN/LAUNCHER activity in Assets/Plugins/Android/AndroidManifest.xml.
using UnityEngine;

namespace AvalonShell
{
    public class AvalonLauncherActivity : com.unity3d.player.UnityPlayerActivity
    {
        // No overrides needed — the shell scene bootstraps via Shell.cs.
    }
}

File: unity/Assets/Plugins/Android/AvalonLauncherActivity.java
// AVALON launcher entry point — Unity 6 disables the stock UnityPlayerActivity when a
// custom main manifest is present, so the launcher must be a custom subclass.
// Documented Unity pattern: Java plugin in Assets/Plugins/Android + MAIN/LAUNCHER in manifest.
package com.bigfoot404.avalon;

public class AvalonLauncherActivity extends com.unity3d.player.UnityPlayerActivity
{
    // Shell scene bootstraps via Shell.cs — no overrides needed.
}
